cloudcontainer.go

```diff
@@ -22,7 +29,7 @@ func (c *cloudContainer) ProviderId() string {
 }
 
 // Address implements CloudContainer.
-func (c *cloudContainer) Address() string {
+func (c *cloudContainer) Address() Address {
```

Comment:
Because it was a pointer, now Address() might return a nil interface.

Reply:
it may, the caller will need to deal with it as it is optional